File: apps/api/domain/workflow.go
package domain

import "time"

type WorkflowStatus string

const (
	WorkflowStatusDraft  WorkflowStatus = "draft"
	WorkflowStatusActive WorkflowStatus = "active"
)

// WorkflowStep is a single execution step stored as part of WorkflowDefinition JSONB.
type WorkflowStep struct {
	ID           string         `json:"id"`
	Name         string         `json:"name"`
	Capability   string         `json:"capability"` // browser|http|llm|notify|code
	Instructions string         `json:"instructions"`
	DependsOn    []string       `json:"depends_on"`
	Config       map[string]any `json:"config,omitempty"`
}

// WorkflowDefinition is AI-generated; stored as JSONB.
type WorkflowDefinition struct {
	Steps []WorkflowStep `json:"steps"`
}

type WorkflowTrigger struct {
	Type   string         `json:"type"` // manual|webhook|schedule
	Config map[string]any `json:"config,omitempty"`
}

type Workflow struct {
	ID          string             `json:"id"`
	UserID      string             `json:"userId"`
	Name        string             `json:"name"`
	Description string             `json:"description"`
	Definition  WorkflowDefinition `json:"definition"`
	Trigger     WorkflowTrigger    `json:"trigger"`
	Status      WorkflowStatus     `json:"status"`
	CreatedAt   time.Time          `json:"createdAt"`
	UpdatedAt   time.Time          `json:"updatedAt"`
}

File: apps/agent/src/contracts/workflow.ts
import { z } from 'zod'

export const WorkflowStepSchema = z.object({
  id:           z.string(),
  name:         z.string(),
  capability:   z.enum(['browser', 'http', 'llm', 'notify', 'code', 'file']),
  instructions: z.string().describe('自然语言描述这步要做什么'),
  depends_on:   z.array(z.string()).default([]),
  config:       z.record(z.unknown()).optional(),
})

export const WorkflowDefinitionSchema = z.object({
  steps: z.array(WorkflowStepSchema).min(1),
})

export type WorkflowStep = z.infer<typeof WorkflowStepSchema>
export type WorkflowDefinition = z.infer<typeof WorkflowDefinitionSchema>

export interface StepResult {
  stepId:   string
  status:   'done' | 'failed'
  output:   string
  data?:    Record<string, unknown>
  error?:   string
}
